src/CMakeLists.txt

```diff
@@ -93,6 +98,9 @@ endif()
 # Include ENTT (header only lib)
 target_include_directories(osp-magnum PRIVATE ../3rdparty/entt/src)
 
+# Include ImPlot (simple library that depends on magnum-integration's imgui)
+target_include_directories(osp-magnum PRIVATE ../3rdparty/implot)
```

Comment:
Doesn't this automatically get picked up when you add the magnum-imgui library to the list of libraries to link to?

Reply:
I believe that implot is separate from ImGui, so you'll have to include it separately

Reply:
That's true, but it's not what I was asking about.

Normally in CMake, when you have a library, even if it's header-only, you create a "library" in cmake that has all the include directories for the library tied to it.

Then, instead of having your own project use "target_include_directories", you simply do "target_link_libraries" and that automatically populates any shared libraries, static libraries, dependencies, and include directories for you.

So my confusion was "Why aren't we using the cmake "library" for imgui/implot, instead of directly adding their include directories?" If we're doing it this way because imgui/implot don't have cmake "libraries", then that's terrible and we should probably send them patches to fix their broken.

Reply:
I can't remember why I did it this way, but generally implot/imgui are meant to be included directly since they're small libraries with no dependencies. Feel free to explain in more detail how I should go about trying what you're suggesting.

Reply:
You should make an "interface" library (or use the one that imgui and implot define, in their own cmake file), and then use "target_link_library(MyImPlotInterfaceLib)" to add that library.

The interface library doesn't have any cpp files or shared librarys / static libraries. It just exists to provide an ergnomic way to refer to a header-only-library and all of it's dependencies.